hw5/main.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional, List
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel, Field
+from uuid import UUID, uuid4
+
+# Модель задачи с Pydantic
+class Task(BaseModel):
+    id: Optional[UUID] = Field(default_factory=uuid4)
+    title: str
+    description: str
+    completed: bool = False
+
+# Модель для обновления задачи
+class TaskUpdate(BaseModel):
+    title: Optional[str]
+    description: Optional[str]
+    completed: Optional[bool]
+
+# Создание экземпляра FastAPI
+app = FastAPI()
+
+# Хранилище задач
+tasks = {}
+
+@app.get("/")
+async def read_root():
+    return {"message": "Hello World"}
+
+@app.get("/tasks", response_model=List[Task])
+async def read_tasks():
+    return list(tasks.values())
+
+@app.get("/tasks/{task_id}", response_model=Task)
+async def read_task(task_id: UUID):
+    if task_id not in tasks:
+        raise HTTPException(status_code=404, detail="Task not found")
+    return tasks[task_id]
+
+@app.post("/tasks", response_model=Task)
+async def create_task(task: Task):
+    tasks[task.id] = task
+    return task
```

Comment:
При тестирование этой конечной точки приходится вручную менять id, Иначе он будет генерироваться всегда одним и тем же. Нужно поменять модель, убрав `default_factory`, а тут добавить
```suggestion
@app.post("/tasks", response_model=Task)
async def create_task(task: Task):
    task.id = uuid4()
    tasks[task.id] = task
    return task
```